include/range/v3/view/cache1.hpp

```diff
@@ -0,0 +1,173 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2014-present
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+#ifndef RANGES_V3_VIEW_CACHE1_HPP
+#define RANGES_V3_VIEW_CACHE1_HPP
+
+#include <range/v3/range_fwd.hpp>
+
+#include <range/v3/detail/range_access.hpp>
+#include <range/v3/iterator/concepts.hpp>
+#include <range/v3/range/access.hpp>
+#include <range/v3/range/concepts.hpp>
+#include <range/v3/range/primitives.hpp>
+#include <range/v3/range/traits.hpp>
+#include <range/v3/utility/semiregular_box.hpp>
+#include <range/v3/view/adaptor.hpp>
+#include <range/v3/view/all.hpp>
+
+namespace ranges
+{
+    /// \addtogroup group-views
+    /// @{
+    template<typename Rng>
+    struct cache1_view : view_facade<cache1_view<Rng>, range_cardinality<Rng>::value>
+    {
+    private:
+        CPP_assert(input_range<Rng>);
```

Comment:
The constraints on `range_value_t<Rng>` (IIRC `copy_constructible` for the `semiregular_box`) are missing here.

Reply:
I should have used `non_propagating_cache` instead, which doesn't have that requirement.

Reply:
Still missing `constructible_from<range_value_t<Rng>, range_reference_t<Rng>>` here.

Reply:
Fine, have it your way.